docs/uml/main/ExtEvSimulationClasses.puml

```diff
@@ -1,223 +1,202 @@
 @startuml
 'https://plantuml.com/class-diagram
 skinparam linetype ortho
+skinparam nodesep 50
+skinparam ranksep 40
 
-package ev-simulation {
-    class ExtLink
-    class ExternalSampleSim
-    class EvModelImpl
-
-    ' adding hidden elements for layout purposes
-    class placeholderA {
-        w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w
-    }
-    hide placeholderA
-
-    class placeholderB {
-        w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w-w
-    }
-    hide placeholderB
+skinparam node {
+    borderColor Transparent
+    fontColor Transparent
+}
 
-    ExtLink -[hidden]> placeholderA
-    placeholderA -[hidden]> ExternalSampleSim
-    ExternalSampleSim -[hidden]> placeholderB
-    placeholderB -[hidden]> EvModelImpl
+package MobilitySimulator {
+    class ExtLink
+    class MobilitySim
+    class ElectricVehicle
 }
 
 package simona-api {
-    ' MIDDLE PART
-    class ExtEvData {
-        ~ LinkedBlockingQueue<EvDataResponseMessageToExt> receiveTriggerQueue
-        - ActorRef dataService
-        - ActorRef extSimAdapter
-        + List<UUID> requestAvailablePublicEvCs()
-        + List<EvModel> sendEvPositions(EvMovementsMessage evMovementsMessage)
-        - void sendData(ExtEvMessage msg)
-        + void queueExtMsg(ExtEvResponseMessage extEvResponse)
-    }
-
-    class ExtSimAdapterData {
-        ~ LinkedBlockingQueue<ExtTrigger> receiveTriggerQueue
-        - ActorRef extSimAdapter
-        + void queueExtMsg(ExtTrigger trigger)
-        + void send(ExtTriggerResponse msg)
-    }
-
-    interface ExtData
 
     abstract class ExtSimulation {
+        - ExtSimAdapterData data
         + void setup(ExtSimAdapterData data, List<ExtData> adapters)
         + void run()
-        # {abstract} List<Long> doActivity(long tick)
+        # {abstract} Optional<Long> doActivity(long tick)
+        # {abstract} Long intitialize()
     }
+    ExtSimulation --r> ExtData
+    MobilitySim -|> ExtSimulation
+
+    interface ExtLinkInterface {
+        + ExtSimulation getExtSimulation()
+        + List<ExtDataSimulation> getExtDataSimulations()
+    }
+
+    ExtLink --|> ExtLinkInterface
 
     interface ExtEvSimulation {
-        + void setExtEvData(ExtEvData evData)
+       + void setExtEvData(ExtEvData evData)
     }
 
+    MobilitySim --|> ExtEvSimulation
+
     interface ExtDataSimulation
 
-    ExternalSampleSim --|> ExtSimulation
-    ExternalSampleSim --|> ExtEvSimulation
+    ExtEvSimulation --|> ExtDataSimulation
```

Comment:
Maybe you can manage to place `ExtEvSimulation` below `ExtDataSimulation`? Not a big deal though.